File: smart_farm/src/api/aiChat.ts
import { get, http } from '@/utils/request'
import type { PageInfo } from '@/api/user'

/**
 * AI对话记录实体，对应 smart_plant 后端 ai_chat 表。
 * 管理端只做查询和删除，不提供人工改写历史对话内容的入口。
 */
export interface AiChat {
  id?: number
  userId?: number
  username?: string
  nickname?: string
  modelId?: number
  modelName?: string
  model?: string
  imageUrl?: string
  userContent?: string
  aiContent?: string
  status?: number
  failReason?: string
  createTime?: string
  updateTime?: string
}

export interface AiChatListParams {
  userId?: number
  username?: string
  modelName?: string
  status?: number
  startDate?: string
  endDate?: string
  pageNum: number
  pageSize: number
}

export type AiChatStatisticsParams = Omit<AiChatListParams, 'pageNum' | 'pageSize'>

export interface AiChatStatistics {
  totalCount?: number
  successCount?: number
  failedCount?: number
  imageCount?: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

/** 分页查询AI对话记录，对接 GET /ai-chat/list。 */
export const listAiChats = (params: AiChatListParams) => {
  return get<ApiResult<PageInfo<AiChat>>>('/ai-chat/list', params as unknown as Record<string, unknown>)
}

/** 查询AI对话统计数据，对接 GET /ai-chat/statistics。 */
export const getAiChatStatistics = (params: AiChatStatisticsParams) => {
  return get<ApiResult<AiChatStatistics>>(
    '/ai-chat/statistics',
    params as unknown as Record<string, unknown>,
  )
}

/** 根据 ID 查询对话详情，供详情弹窗展示完整消息。 */
export const getAiChatById = (id: number) => {
  return get<ApiResult<AiChat>>(`/ai-chat/${id}`)
}

/** 删除单条对话记录。 */
export const deleteAiChat = (id: number) => {
  return http<ApiResult<void>>({ url: `/ai-chat/${id}`, method: 'delete' })
}

/** 批量删除对话记录。 */
export const batchDeleteAiChats = (ids: number[]) => {
  return http<ApiResult<number>>({ url: '/ai-chat/batch', method: 'delete', data: ids })
}

File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/AiChat.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * AI对话记录实体类。
 *
 * <p>该类映射 ai_chat 表，用于保存用户输入、可选附件、模型回复和调用状态。</p>
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AiChat {

    /** 对话记录主键ID。 */
    private Long id;

    /** 发起对话的用户ID。 */
    private Long userId;

    /** 用户名，列表查询时通过 user 表关联回填。 */
    private String username;

    /** 用户昵称，列表查询时通过 user 表关联回填。 */
    private String nickname;

    /** 使用的模型配置ID，对应 model.id。 */
    private Long modelId;

    /** 模型展示名称，列表查询时通过 model 表关联回填。 */
    private String modelName;

    /** 实际调用的模型名称，列表查询时通过 model 表关联回填。 */
    private String model;

    /** 兼容数据库既有字段：保存图片或文档附件地址，纯文本咨询时为空。 */
    private String imageUrl;

    /** 文档附件的原始文件名，用于刷新历史后保持用户上传时的名称。 */
    private String fileName;

    /** 用户发送的咨询内容。 */
    private String userContent;

    /** AI返回的回复内容。 */
    private String aiContent;

    /** 调用状态，1成功、2失败。 */
    private Integer status;

    /** 失败原因，仅 status=2 时记录。 */
    private String failReason;

    /** 创建时间，由数据库默认写入。 */
    private LocalDateTime createTime;

    /** 更新时间，由数据库自动维护。 */
    private LocalDateTime updateTime;
}
